pkg/gallery/gallery.go

```diff
@@ -132,6 +134,61 @@ func AvailableGalleryModels(galleries []Gallery, basePath string) ([]*GalleryMod
 	return models, nil
 }
 
+// This is ***NEVER*** going to be perfect or finished.
+// This is a BEST EFFORT function to surface known-vulnerable models to users.
+func SafetyScanGalleryModels(galleries []Gallery, basePath string) error {
+	galleryModels, err := AvailableGalleryModels(galleries, basePath)
+	if err != nil {
+		return err
+	}
+	for _, gM := range galleryModels {
+		if gM.Installed == true {
+			err = errors.Join(err, SafetyScanGalleryModel(gM))
+		}
+	}
+	return err
+}
+
+func SafetyScanGalleryModel(galleryModel *GalleryModel) error {
+	for _, file := range galleryModel.AdditionalFiles {
+		switch {
+		case strings.HasPrefix(file.URI, downloader.HuggingFacePrefix):
+			templatedUrl := downloader.ConvertURL(file.URI)
+			parts := strings.Split(templatedUrl, "/")
+			repoURL := fmt.Sprintf("https://huggingface.co/%s/%s", parts[4], parts[5])
+			resp, err := http.Get(repoURL)
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode != 200 {
+				return fmt.Errorf("[SecurityScan] received an unexpected status code %d while querying %q", resp.StatusCode, repoURL)
+			}
+			doc, err := goquery.NewDocumentFromReader(resp.Body)
+			if err != nil {
+				return err
+			}
+			// TODO: Is there a more elegant method than Map here I'm missing?
+			parsed := doc.Find("div.alert-danger div p").Map(func(i int, s *goquery.Selection) string {
```

Comment:
Perhaps now that you've added `ci/modelslist.go` maybe it makes sense to put this check into that code there - it's better for the brittle check to prevent it from displaying than from installing I guess? I think that there may still be _a role_ for the cli `secscan` command - we need some way for users to be able to check models they have already installed against _current_ repo warnings, in case things were discovered post install. Currently we don't trace the provenance of installed model configs back to a gallery source, so that's always going to be somewhat best effort mapping.

Absolutely no disagreement that this is fragile and brittle - just think we should have _some_ line of defense here.

Reply:
Ah, I also see the bash script `.github/checksum_checker.sh` you may have been referring to. It may be equally simple to toss it into that stage as well - I'll do some prototyping around there, as I think that's a better separation of concerns from modelslist.go

Reply:
One bit of bad news: I poked around the `huggingface_hub` python library that script uses, since I have a little experience with it from my prior attempts at an automated gallery. None of their api functions expose the fact that the model is vulnerable... so crappy brittle tag parsing may still be in the cards :\